api/v1alpha1/machine.go

```diff
@@ -49,6 +49,25 @@ const (
 type MachineSpec struct {
 	// Connection contains connection data for a Baseboard Management Controller.
 	Connection Connection `json:"connection"`
+
+	// ProviderOptions contains provider specific options.
+	// +optional
+	ProviderOpts ProviderOpts `json:"providerOpts,omitempty"`
+}
+
+type ProviderOpts struct {
```

Comment:
Do you mean pointers should be used here?

Reply:
That's certainly what I would do, though I know from previous conversations there's hesitation around that. I was more curious how you'd solve this problem though, because from what I can tell its possible someone accidentally leaves out configuration and I don't think we could tell. 

For example, its self evident with:

```
v := ProviderOpts{
   IntelAMT: &IntelAMTOpts{...}
}
```

We can tell they specified IntelAMT and can validate that data structure. Without the pointer, I don't know how we can differentiate erroneous input from "unspecified". As I've shared before, I certainly prefer the clarity of pointers at the expense of additional checks in the same way I prefer the clarity of Go's explicit error returns.

Reply:
Yup, all good. Made them pointers.